ibis/backends/flink/compiler.py

```diff
@@ -0,0 +1,642 @@
+"""Flink Ibis expression to SQL compiler."""
+
+from __future__ import annotations
+
+from functools import singledispatchmethod
+
+import sqlglot as sg
+import sqlglot.expressions as sge
+from sqlglot.dialects import Hive
+from sqlglot.dialects.dialect import rename_func
+
+import ibis.common.exceptions as com
+import ibis.expr.datatypes as dt
+import ibis.expr.operations as ops
+from ibis.backends.base.sqlglot.compiler import STAR, SQLGlotCompiler, paren
+from ibis.backends.base.sqlglot.datatypes import FlinkType
+from ibis.backends.base.sqlglot.rewrites import (
+    exclude_unsupported_window_frame_from_ops,
+    exclude_unsupported_window_frame_from_rank,
+    exclude_unsupported_window_frame_from_row_number,
+    rewrite_first_to_first_value,
+    rewrite_last_to_last_value,
+    rewrite_sample_as_filter,
+)
+
+
+class Flink(Hive):
```

Comment:
Shouldn't this go to sqlglot/dialects.py?

Reply:
Yep, thanks! I started this PR before I moved them. Moving now.

Reply:
Moved, thanks!